File: haskell/comments.hs
#!/usr/bin/env runhaskell

-- Allows command line flags to derive Typeable and Data
{-# LANGUAGE ScopedTypeVariables, DeriveDataTypeable #-}

-- Single line comment

{- Multi
line
comment
-}

-- Type signature
main :: IO ()
main = putStrLn "Bye"

File: lua/comments.lua
#!/usr/bin/env lua

-- Single line comment

--[[
  Multiline
  comment
]]

Comments = {}

local main
main = function(arg)
  print("Bye")
end

if type(package.loaded[(...)]) ~= "userdata" then
  main(arg)
else
  module(..., package.seeall)
end
